src/ethereum_test_forks/transition_base_fork.py

```diff
@@ -26,24 +26,43 @@ def transitions_from(cls) -> Fork:
         raise Exception("Not implemented")
 
 
-def transition_fork(to_fork: Fork):
+def base_fork_abstract_methods() -> List[str]:
+    """
+    Returns a list of all abstract methods that must be implemented by a fork.
+    """
+    return list(getattr(BaseFork, "__abstractmethods__"))
+
+
+def transition_fork(to_fork: Fork, at_block: int = 0, at_timestamp: int = 0):
     """
     Decorator to mark a class as a transition fork.
     """
 
     def decorator(cls) -> Type[TransitionBaseClass]:
         transition_name = cls.__name__
+        from_fork = cls.__bases__[0]
+        assert issubclass(from_fork, BaseFork)
+
+        class NewTransitionClass(cls, TransitionBaseClass, BaseFork):  # type: ignore
+            pass
+
+        NewTransitionClass.name = lambda: transition_name  # type: ignore
+
+        for method_name in base_fork_abstract_methods():
+
+            def _method(cls, block_number: int, timestamp: int, method_name=method_name):
+                if block_number >= at_block and timestamp >= at_timestamp:
+                    return getattr(to_fork, method_name)(block_number, timestamp)
+                return getattr(from_fork, method_name)(block_number, timestamp)
 
-        class NewTransitionClass(cls, TransitionBaseClass):  # type: ignore
-            @classmethod
-            def name(cls) -> str:
-                """
-                Returns the name of the transition fork.
-                """
-                return transition_name
+            setattr(
+                NewTransitionClass,
+                method_name,
+                classmethod(_method),
+            )
```

Comment:
The need for a dynamic method was because the methods were neither from one fork or the other, but it performs the block/timestamp check before calling either of them.

I think it should be possible to override the methods automatically when the subclass is created, and we could definitely explore if `__init_subclass__` could suit our needs for this.